### PR TITLE
Excalibur vds

### PR DESCRIPTION
**Pull request by gilesknap**
Tests on python 3 are now passing.

### CODE REVIEW

**Comment by gilesknap** [comment]
Hi have to leave now but have noticed that this has pulled in some of Toms changes and not sure why.

File: requirements/test.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@ plop==0.3.0
 pytest>=3.2.0
 pytest-cov<=2.6.0
 h5py>=2.8.0
-vdsgen>=0.3
-annotypes==0.9
+annotypes>=0.9
```
Comment (thomascobb) [inline requirements/test.txt]:
```suggestion
annotypes==0.9
```
Need to leave this pinned as new versions of annotypes will break this...

File: malcolm/modules/ADOdin/parts/odinwriterpart.py [hunk 2]
```diff
@@ -80,45 +238,36 @@ def configure(self,
                   completed_steps,  # type: scanning.hooks.ACompletedSteps
                   steps_to_do,  # type: scanning.hooks.AStepsToDo
                   generator,  # type: scanning.hooks.AGenerator
-                  fileDir='/tmp',  # type: AFileDir
-                  fileName="odin.hdf",  # type: AFileName
+                  fileDir,  # type: AFileDir
+                  formatName="odin",  # type: AFormatName
+                  fileTemplate="%s.hdf",  # type: AFileTemplate
```
Comment (thomascobb) [inline malcolm/modules/ADOdin/parts/odinwriterpart.py]:
```suggestion
                  fileTemplate="%s.h5",  # type: AFileTemplate
```

File: setup.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def get_version():
         raise RuntimeError('Unable to find version string in {0}.'
                            .format(VERSION_FILE))
 
-install_requires = ['numpy', 'scanpointgenerator>=1.5', 'annotypes']
+install_requires = ['numpy', 'scanpointgenerator>=1.5', 'annotypes', 'vdsgen']
```
Comment (thomascobb) [inline setup.py]: Not install_requires. We need to define some optional requires targets for hdf5, pvaccess, etc. Cothread does need to be in this list now though...

File: tests/test_modules/test_ADOdin/test_odinwriterpart.py [hunk 4]
```diff
@@ -1,58 +1,202 @@
...
+        assert detector_group.attrs['y_set_indices'] == '0'
+        assert detector_group.attrs['x_set_indices'] == '1'
+
+        # Check _set datasets
+        # N.B. units are encoded as ASCII in the original file, so come
+        # back as type byte in Python 3
+        stage1_x_set_dataset = vds_file['/entry/detector/x_set']
+        assert stage1_x_set_dataset[0] == 0
+        assert stage1_x_set_dataset[1] == 2
+        assert stage1_x_set_dataset.attrs['units'] == b'mm'
```
Comment (thomascobb) [inline tests/test_modules/test_ADOdin/test_odinwriterpart.py]: Is this really a bytes literal? Does h5py not convert to Python3 string?

**Comment by thomascobb** [comment]
My commits are because of the re-rebase that we did, Github keeps track of things that git forget about...

File: tests/test_modules/test_ADOdin/test_odinwriterpart.py (hunk 4 above)
Comment (gilesknap) [reply tests/test_modules/test_ADOdin/test_odinwriterpart.py]: It seems I lied about Python 3 tests working. In Python3 this is str but in Python2 it is bytes. Will change the test to case the value to str.

File: setup.py (hunk 3 above)
Comment (gilesknap) [reply setup.py]: I've removed vdsgen and added cothread. Are the optional targets a future TODO? The imalcolm entry point does not list any of the extras_requires entries, so at present setup.py does not seem to bring in any extra dependencies.

**Pull request merged**